### ISSUE TITLE
Is your attack method a new one or a reproduction of Song's paper?

### ISSUE THREAD

**Issue by 2016312357** [The-MinGo/NISL-FL#3]
Hello, I've been interested in membership inference attacks against FL. And I found your code on Github. I'd appreciate that if you could give a more detailed description of your attack method, since the code is a little hard to follow.

**Comment by gongzhimin** [comment]
Hello, @2016312357. Thank you for your attention!
The membership inference attack in this repo is from Shokri's paper [1]. And [here](https://github.com/privacytrustlab/ml_privacy_meter) the code, which is similar to [Song's implementation](https://github.com/csong27/membership-inference) [2].
I'm sorry there are still some hidden faults in the `master` branch as I want to refactor Shokri's code. But the branch [`old-version`](https://github.com/gongzhimin/Membership-Inference-Attack-in-Federated-Learning/tree/old-version) is okay.

[1] Comprehensive Privacy Analysis of Deep Learning: Passive and Active White-box Inference Attacks against Centralized and Federated Learning (https://arxiv.org/abs/1812.00910).
[2] Membership Inference Attack against Machine Learning Models (http://ieeexplore.ieee.org/document/7958568/).

**Comment by macknight** [comment]
Hi,
Is the master branch runnable now?

BR

**Comment by gongzhimin** [comment]
Hallo, @macknight.
The **master** branch is runnable, but the results are uncorrected yet. >~< Maybe because I defined the structures of models unsuitably. 
Sorry, I'm writing a paper currently, and unavailable to fix these long-ago faults in this branch. But the **old-version** branch is ok.

Sincerely

**Comment by macknight** [comment]
Thanks
